File: Blind 75/Arrays/Product of Array Except Self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] ansarr = new int[nums.length];
        ansarr[0] = 1;
        int R = 1;
        //prefix
        for (int i = 1; i < nums.length; i++) {
            ansarr[i] = nums[i - 1] * ansarr[i - 1];
        }
        //suffix
        for (int j = nums.length - 1; j > -1; j--) {
            ansarr[j] *= R;
            R *= nums[j];
        }
        return ansarr;
    }
}

File: Python Blind 75/Arrays/Product of Array Except Self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        # Time: O(n), Space: O(1)
        ans = [1] * len(nums)
        prefix = 1
        # prefix
        for i in range(len(nums)):
            ans[i] = prefix
            prefix *= nums[i]
        # postfix
        postfix = 1
        for i in range(len(nums) - 1, -1, -1):
            ans[i] *= postfix
            postfix *= nums[i]

        return ans
        